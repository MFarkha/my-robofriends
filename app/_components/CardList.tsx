import React from "react";
import Card from "./Card";
import { IRobot } from "../page";

type CardListProps = {
  error: string;
  robots: Array<IRobot>;
};

const CardList = ({ error, robots }: CardListProps) => {
  if (error.length !== 0) {
    throw new Error();
  }
  return (
    <div>
      {robots.map((robot) => {
        return <Card robot={robot} key={robot.id} />;
      })}
    </div>
  );
};

export default CardList;
