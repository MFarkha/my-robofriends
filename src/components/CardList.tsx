import Card from './Card';
import { IRobot } from '../app/store';

type CardListProps = {
    robots: Array<IRobot>
};

const CardList = ({ robots }: CardListProps) => {
    return (
        <div>
            {
                robots.map((robot) => {
                    return (
                        <Card robot={robot} key={robot.id}/>
                    );
                })
            }
        </div>
    );
}

export default CardList;