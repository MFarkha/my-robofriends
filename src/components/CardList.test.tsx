import { shallow } from "enzyme";
import CardList from "./CardList";

it("expects to render the CardList component", () => {
  const mockRobots = [
    {
      key: 10,
      id: 1,
      name: "John Snow",
      email: "john@example.com",
    },
    {
      key: 11,
      id: 2,
      name: "Kelly Gale",
      email: "kelly@example.com",
    },
  ];
  const wrap = shallow(<CardList error="" robots={mockRobots} />);
  expect(wrap).toMatchSnapshot();
});
