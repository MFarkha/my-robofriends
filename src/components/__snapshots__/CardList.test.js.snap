// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to render the CardList component 1`] = `
<div>
  <Card
    email="john@example.com"
    id={1}
    key="1"
    name="John Snow"
  />
  <Card
    email="kelly@example.com"
    id={2}
    key="2"
    name="Kelly Gale"
  />
</div>
`;
